retrofitsample/src/test/java/test/Authenticator.kt

```diff
@@ -0,0 +1,360 @@
+                        "  \"expires_in\":3000,\n" +
+                        "  \"token_type\":\"Bearer\"\n" +
+                        "}")
+              }
+              else if (params["refresh_token"] == "123" && params["grant_type"] == "refresh_token")
+              {
+                MockResponse().setResponseCode(200)
+                    .setBody("{\n" +
+                        "  \"access_token\":\"abc\",\n" +
+                        "  \"refresh_token\":\"123\",\n" +
+                        "  \"expires_in\":3000,\n" +
+                        "  \"token_type\":\"Bearer\"\n" +
+                        "}")
+              }
+              else
+              {
+                MockResponse().setResponseCode(403)
+                    .setBody("{\n" +
+                        "  \"status_code\": 403,\n" +
+                        "  \"message\":\"Forbidden\"\n" +
+                        "}")
+              }
+            }
+            else
+            {
+              if (params["username"] == "user" && params["password"] == "pwd" && params["grant_type"] == "password")
+              {
+                MockResponse().setResponseCode(200)
+                    .setBody("{\n" +
+                        "  \"access_token\":\"abc\",\n" +
+                        "  \"refresh_token\":\"123\",\n" +
+                        "  \"expires_in\":3000,\n" +
+                        "  \"token_type\":\"Bearer\"\n" +
+                        "}")
+              }
+              else if (params["refresh_token"] == "123" && params["grant_type"] == "refresh_token")
+              {
+                MockResponse().setResponseCode(200)
+                    .setBody("{\n" +
+                        "  \"access_token\":\"def\",\n" +
+                        "  \"refresh_token\":\"456\",\n" +
+                        "  \"expires_in\":3000,\n" +
+                        "  \"token_type\":\"Bearer\"\n" +
+                        "}")
+              }
+              else
+              {
+                MockResponse().setResponseCode(403)
+                    .setBody("{\n" +
+                        "  \"status_code\": 403,\n" +
+                        "  \"message\":\"Forbidden\"\n" +
+                        "}")
+              }
+            }
+          }
+
+          if (request.path == "/info/")
+          {
+            if (request.getHeader("Authorization")?.isNotBlank() == true)
+            {
+              when (behavior)
+              {
+                Companion.ServerBehavior.NORMAL                  ->
+                {
+                  if (request.getHeader("Authorization") == "Bearer abc")
+                    return MockResponse().setResponseCode(200).setBody("{\"info\":{\"name\":\"Lucas Albuquerque\",\"age\":\"21\",\"gender\":\"male\"}}")
+                  else
+                    return MockResponse().setResponseCode(401)
+                        .setBody("{\n" +
+                            "  \"status_code\": 401,\n" +
```

Comment:
For a better readability of the JSON, you can use the 3 quotes system """ """

```suggestion
"""{
                    "status_code": 403
                    "message": "Forbidden"
                    }"""
```

Reply:
refactored